src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1590,6 +1590,8 @@ class DocBaseViewer extends BaseViewer {
                 : nextMode,
         );
         this.processAnnotationModeChange(nextMode);
+
+        this.updateDiscoverabilityResinTag();
```

Comment:
Should we call this function for every FSM transition? There're also transitions in `handleAnnotationCreateEvent`, should we call this there? Is it adding a state change callback in FSM a good idea?

Reply:
It seems okay to me. That's basically how redux works with `subscribe`, so probably fine?

Reply:
The reality is, we only really care to change the resin attribute when toggling in and out of explicit modes, but moving this to some sort of subscribe callback would be more comprehensive. If we introduce a subscribe callback, we could probably move the calls to `processAnnotationModeChange` there as well. Any objections to doing this in a follow up refactor?

Reply:
I had to do it anyway to account for the differences between image discoverability and doc discoverability